src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -394,7 +482,21 @@ internal static class EventLogEntryCollectionExtensions
     {
         internal static EventLogEntry LastOrDefault(this EventLogEntryCollection elec)
         {
-            return elec.Count > 0 ? elec[elec.Count - 1] : null;
+            bool entryRetrieved = true;
+            EventLogEntry eventLogEntry = null;
+            while (entryRetrieved)
+            {
+                try
+                {
+                    eventLogEntry = elec.Count > 0 ? elec[elec.Count - 1] : null;
+                    entryRetrieved = false;
+                }
+                catch (Win32Exception)
```

Comment:
Did you repro the Win32EXception locally, or just the issue of the events not immediately being represented in .Count ?  In other words do we know that retries will help with the exception?